src/utils/transactionUtils/transactionUtils.ts

```diff
@@ -101,72 +102,32 @@ export function isInteractionCompatible({
 export function getTransactionPayloadByInteractionType(
   contractType: ContractType,
   interactionType: AntInteraction | RegistryInteraction,
-  data: string | string[],
+  data?: string | string[],
 ): TransactionData | undefined {
   try {
-    const payload: any = {};
+    if (!data) {
+      throw new Error(
+        'No data provided, data is required to build the payload',
+      );
+    }
     const txData = typeof data === 'string' ? [data] : [...data];
-
+    const payload = mapTransactionDataKeyToPayload(
+      contractType,
+      interactionType,
+      txData,
+    );
+    // overrides
     if (
       contractType == CONTRACT_TYPES.ANT &&
       isAntInteraction(interactionType)
     ) {
-      if (
-        !isInteractionCompatible({
-          contractType,
-          interactionType,
-          functionName:
-            TRANSACTION_DATA_KEYS[contractType][interactionType].functionName,
-        })
-      ) {
-        throw new Error(`Interaction is not compatible`);
-      }
-      payload.functionName =
-        TRANSACTION_DATA_KEYS[contractType][interactionType].functionName;
       switch (interactionType) {
-        case ANT_INTERACTION_TYPES.SET_TICKER:
-        case ANT_INTERACTION_TYPES.SET_NAME: {
-          TRANSACTION_DATA_KEYS[contractType][interactionType].keys.forEach(
-            (key: string, index) => {
-              if (!txData[index]) {
-                throw new Error(
-                  `Missing key (${key}) from transaction data in the url. This may be due to the order of the data, the current order is [${txData}], the correct order is [${TRANSACTION_DATA_KEYS[contractType][interactionType].keys}]`,
-                );
-              }
-              payload[key] = txData[index];
-            },
-          );
-        }
-      }
-    }
-
-    if (
-      contractType == CONTRACT_TYPES.REGISTRY &&
-      isRegistryInteraction(interactionType)
-    ) {
-      if (
-        !isInteractionCompatible({
-          contractType,
-          interactionType,
-          functionName:
-            TRANSACTION_DATA_KEYS[contractType][interactionType].functionName,
-        })
-      ) {
-        throw new Error(`Interaction is not compatible`);
-      }
-      payload.functionName =
-        TRANSACTION_DATA_KEYS[contractType][interactionType].functionName;
-      TRANSACTION_DATA_KEYS[contractType][interactionType].keys.forEach(
-        (key, index) => {
-          if (!txData[index]) {
-            // if missing transaction data from the url, throw an error
-            throw new Error(
-              `Missing key (${key}) from transaction data in the url. This may be due to the order of the data, the current order is [${txData}], the correct order is [${TRANSACTION_DATA_KEYS[contractType][interactionType].keys}]`,
-            );
+        case ANT_INTERACTION_TYPES.SET_TARGET_ID:
+          {
+            payload.subDomain = '@';
```

Comment:
so just one case in the switch?

Reply:
yes, the switch is now for overrides on special interaction types